File: src/main/java/sharev/domain/connection/event/ShowCardEventListener.java
package sharev.domain.connection.event;

import lombok.RequiredArgsConstructor;
import lombok.extern.slf4j.Slf4j;
import org.springframework.context.event.EventListener;
import org.springframework.scheduling.annotation.Async;
import org.springframework.stereotype.Component;
import sharev.domain.connection.service.ConnectionService;

@Slf4j
@Component
@RequiredArgsConstructor
public class ShowCardEventListener {

    private final ConnectionService connectionService;

    @Async
    @EventListener
    public void connect(ShowCardEvent showCardEvent) {
        try {
            connectionService.connect(showCardEvent.eventId(), showCardEvent.accountId(), showCardEvent.targetCardId());

        } catch (Exception e) {
            log.error("Failed to process ShowCardEvent - eventId: {}, accountId: {}, targetCardId: {}",
                    showCardEvent.eventId(), showCardEvent.accountId(), showCardEvent.targetCardId(), e);
        }
    }
}

File: src/main/kotlin/sharev/connection/adapter/inbound/event/ShowCardEventListener.kt
package sharev.connection.adapter.inbound.event

import org.slf4j.LoggerFactory
import org.springframework.context.event.EventListener
import org.springframework.scheduling.annotation.Async
import org.springframework.stereotype.Component
import sharev.card.domain.event.ShowCardEvent

@Component
class ShowCardEventListener(
    private val connectCardsUseCase: sharev.connection.application.port.inbound.usecase.ConnectCardsUseCase,
) {
    private val log = LoggerFactory.getLogger(javaClass)

    @Async
    @EventListener
    fun connect(showCardEvent: ShowCardEvent) {
        try {
            connectCardsUseCase.connect(
                _root_ide_package_.sharev.connection.application.port.inbound.command.ConnectCardsCommand(
                    gatheringId = showCardEvent.eventId,
                    accountId = showCardEvent.accountId,
                    targetCardId = showCardEvent.targetCardId,
                )
            )
        } catch (e: Exception) {
            log.error(
                "Failed to process ShowCardEvent - eventId: {}, accountId: {}, targetCardId: {}",
                showCardEvent.eventId,
                showCardEvent.accountId,
                showCardEvent.targetCardId,
                e,
            )
        }
    }
}
